Sources/MongoSwift/SDAM.swift

```diff
@@ -233,25 +208,20 @@ public struct TopologyDescription {
         return self.servers[0].setName
     }
 
-    /// The largest setVersion ever reported by a primary.
-    public var maxSetVersion: Int64?
-
-    /// The largest electionId ever reported by a primary.
-    public var maxElectionId: ObjectId?
-
-    /// The servers comprising this topology. By default, no servers.
-    public var servers: [ServerDescription] = []
-
-    /// For single-threaded clients, indicates whether the topology must be re-scanned.
-    public let stale: Bool = false // currently, this will never be set
+    /// The servers comprising this topology.
+    public let servers: [ServerDescription]
 
-    /// Exists if any server's wire protocol version range is incompatible with the client's.
-    public let compatibilityError: MongoError? = nil // currently, this will never be set
+    /// Whether every server's wire protocol version range is compatible with MongoSwift's.
+    public var compatible: Bool {
+        return self.servers.allSatisfy {
+            $0.maxWireVersion >= MIN_SUPPORTED_WIRE_VERSION
```

Comment:
This logic is incorrect. From the spec:
```
A ServerDescription which is not Unknown is incompatible if:

minWireVersion > clientMaxWireVersion, or
maxWireVersion < clientMinWireVersion
```

Here we are not ignoring `Unknown` servers, nor are we checking the `minWireVersion` requirements. Additionally, the reason the error itself is stored is because it will likely change based on what part of this is incompatible, its unclear you'd have that information where you eventually report the error to the user.

Reply:
For now I've just removed it.